src/handlers/initiateDataRequest.ts

```diff
@@ -1,18 +1,28 @@
 import { APIGatewayProxyResult, APIGatewayProxyEvent } from 'aws-lambda'
 import { initiateDataTransfer } from '../services/initiateDataTransfer'
 import { updateZendeskTicket } from '../services/updateZendeskTicket'
+import { isValidSignature } from '../services/validateRequestSource'
 import { validateZendeskRequest } from '../services/validateZendeskRequest'
 import { DataRequestParams } from '../types/dataRequestParams'
 import { ValidatedDataRequestParamsResult } from '../types/validatedDataRequestParamsResult'
 
 export const handler = async (
   event: APIGatewayProxyEvent
 ): Promise<APIGatewayProxyResult> => {
+  const headerSignature = event.headers['X-Zendesk-Webhook-Signature'] as string
+  const timestamp = event.headers[
+    'X-Zendesk-Webhook-Signature-Timestamp'
+  ] as string
+  if (
+    !(await isValidSignature(headerSignature, event.body as string, timestamp))
+  ) {
+    return await handleInvalidRequest()
```

Comment:
Since you're doing something very different in the two cases, I would just add another method to handle the case of the signature being invalid. `handleInvalidRequest` has become a bit confusing now, so that would help with that too.

Reply:
I also wonder if to avoid clutter here, it would be easier if `isValidSignature` just accepted an `APIGatewayProxyEvent` and you can then handle all the stuff around pulling the headers out and checking they're set, and whether the body is missing or not in that function, rather than inflating the code in this handler.

Reply:
have addressed these in latest commit. Please review